### PR TITLE
Update comrak to 0.2.3, use ext_header_ids

### LINKED ISSUE #1037: Anchor links on READMEs don't work
GitHub automatically adds anchor links to header tags, allowing links to sections inside of a README. It would be nice if this also worked on crates.io.

Compare clap on [crates.io](https://crates.io/crates/clap) to [GitHub](https://github.com/kbknapp/clap-rs).

### PR DESCRIPTION
**Pull request by kivikakk**
Fixes #1037. Use latest comrak, and use the header IDs extension to generate GitHub-compatible anchors.

/cc @kejadlen @treiff

### CODE REVIEW

**Comment by kivikakk** [comment]
It occurs to me this won't _quite_ fix #1023; the anchors still won't work because we prefix `"user-content-"` to IDs to prevent nasty name collisions and subsequent security issues, but we need to still support `href`s without this prefix (because users don't expect to see `#user-content-…` in their address bar).

I'll come up with an appropriate solution shortly; on GitHub we use JavaScript which fires on hash change to search for and scroll to the item with the `"user-content-"` prefix.

**Comment by kivikakk** [comment]
So, to prevent ID collisions, we'll also need to sanitize the IDs not coming from comrak-outputted header tags; I've proposed https://github.com/notriddle/ammonia/pull/66 to give us the machinery to do that.

Finally, we'll need either the JavaScript portion to scroll the appropriate tag into view (I could throw this together easily based on what we're using here), or we'd need to change all relative `href`s. The former is a bit prettier since it means we don't end up with `"user-content-"` or similar in the address bar.

**Comment by kivikakk** [comment]
The JavaScript side of the solution is now in this PR; ready for your thoughts/comments!

File: src/render.rs [hunk 1]
```diff
@@ -53,8 +53,7 @@ impl<'a> MarkdownRenderer<'a> {
             .cloned()
             .collect();
         let tag_attributes = [
-            ("a", ["href", "target"].iter().cloned().collect()),
-            ("code", ["class"].iter().cloned().collect()),
```
Comment (carols10cents) [inline src/render.rs]: hm, does `allowed_classes` make this line unnecessary...? I guess so since the tests pass!

**Comment by carols10cents** [comment]
Looks great, thank you!!! I'm going to review your other readme-related PRs then deploy and regenerate the readmes all at once, that'll probably take me a few days :) Thank you so much for taking care of all these things!!!

bors: r+

File: src/render.rs (hunk 1 above)
Comment (kivikakk) [reply src/render.rs]: Indeed, it even became required in the newer `ammonia`: https://github.com/notriddle/ammonia/blob/2581eec04ef98419fd202e0dd4d8997c9dfdb06e/src/lib.rs#L1030-L1037 (you're allowed to use `allowed_classes` or set `class` as an allowed tag, but not both)

**Comment by kivikakk** [comment]
Thanks so much!

**Pull request merged**